api/mqtt/connector.js

```diff
@@ -54,18 +55,18 @@ function Broker(conf) {
         var retries = 0;
         try {
             if ((me.client instanceof mqtt.MqttClient) === false) {
-                if (me.secure === false) {
-                    me.client = mqtt.connect('mqtt://' + me.host + ':' + me.port, {
-                        username: me.deviceInfo.device_id,
-                        password: me.deviceInfo.device_token
-                    });
-                } else {
-                    me.client = mqtt.connect('mqtts://' + me.host + ":" + me.port, {
-                        username: me.deviceInfo.device_id,
-                        password: me.deviceInfo.device_token,
-                        rejectUnauthorized: false
-                    });
+                var protocol = 'mqtt://';
+                if (me.secure) {
+                    protocol = 'mqtts://'
+                }
+                if (me.protocol) {
+                    protocol = me.protocol + '://';
                 }
```

Comment:
I think we should do some sanity checking here to make sure protocol isn't something stupid. I assume there should be 3 valid options, mqtt, mqtts and ws?

Reply:
You can ignore this if you agree with my comments above :)